lite2/client.go

```diff
@@ -714,39 +714,52 @@ func (c *Client) bisection(
 	newVals *types.ValidatorSet,
 	now time.Time) error {
 
+	type HeaderSet struct {
+		sh  *types.SignedHeader
+		val *types.ValidatorSet
+	}
+
 	var (
+		headerCache = []HeaderSet{{newHeader, newVals}}
```

Comment:
can't we just use `map[height]{Header, Vals}`? how's having a separate depth variable  better?

Reply:
because we are jumping from header to header we just want a queue not a map, having height is not useful, we just increment the array

Reply:
I guess what I was trying to say code is hard to understand without a visual. Can we at least add the above image to doc.go file or somewhere else. or create an asci version

Reply:
I will try add some more clearer descriptions to doc.go and to the actual bisection method